pkg/server/server.go

```diff
@@ -60,6 +60,10 @@ func StartServer(ctx context.Context, config *Config) error {
 		return errors.Wrap(err, "starting tls server")
 	}
 
+	for _, hook := range config.StartupHooks {
+		hook(config.ControlConfig)
```

Comment:
I think the hook function should return an error and it should be wrapped and return if not nil

Reply:
oh, the function already does but this doesnt check them